File: src/decorator/java/com/my/SmsNotificationSenderDecorator.java
package src.decorator.java.com.my;

public class SmsNotificationSenderDecorator extends NotificationSenderDecorator {
    public SmsNotificationSenderDecorator(NotificationSender sender) {
        super(sender);
    }

    @Override
    public void sendNotification() {
        System.out.println("Gotta send SMS notification");
        super.sendNotification();
        System.out.println("SMS notification sent!");
    }
}

File: src/decorator/python/SmsNotificationSenderDecorator.py
from src.decorator.python.NotificationSenderDecorator import NotificationSenderDecorator


class SmsNotificationSenderDecorator(NotificationSenderDecorator):
    def sendNotification(self):
        print("Gotta send SMS notification")
        self.sender.sendNotification()
        print("SMS notification sent!")
